core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -820,12 +855,12 @@ private static RexLiteral toLiteral(RelDataType type, Comparable<?> value) {
     case NULL:
       return new RexLiteral(null, type, typeName);
     case INTERVAL_WEEK:
-      long weekMillis =
-          SqlParserUtil.intervalToMillis(
+      long weekNanos =
+          SqlParserUtil.intervalToNanos(
               literal,
               castNonNull(type.getIntervalQualifier()),
               typeSystem);
-      return new RexLiteral(BigDecimal.valueOf(weekMillis), type, typeName);
+      return new RexLiteral(BigDecimal.valueOf(weekNanos), type, typeName);
```

Comment:
Is there a reason you changed week to operate on nanseconds? This should be fine, 2^63 > 1 week in nanseconds by about 4 powers of 10, so I don't think overflow is a major concern, but I want to know why you did this.

Reply:
I refactored the function intervalToMillis to intervalToNanos. I can add intervalToMillis if needed